common/datavalues/src/types/deserializations/variant.rs

```diff
@@ -113,10 +113,8 @@ impl TypeDeserializer for VariantDeserializer {
         _format: &FormatSettings,
     ) -> Result<()> {
         self.buffer.clear();
-        reader.read_quoted_text(&mut self.buffer, b'"')?;
-
-        let val = serde_json::from_slice(self.buffer.as_slice())?;
-        self.builder.append_value(val);
+        let val = reader.read_json_value()?;
```

Comment:
csv has it own escape rule,  I suggest read this field as a csv string (unescaped) then decode with json

Reply:
In function `read_csv_string`, we use a comma as delimiter to split the string, but the comma may be part of the `JSON` value, for example `[1,2,3]`. I'm afraid the `JSON` value can't be parsed correctly from the `CSV` string.

Reply:
Any text-based format has its own way of dealing with it using escape and/or escape.
 
the outer layer  CSV should only know the field is a string/number/date.., nothing else, no use to invent our own CSV that can not be parsed by other CSV readers.

 by default , if there is `,` in the string, the CSV need to enclose it with a quote

https://docs.python.org/3/library/csv.html
https://clickhouse.com/docs/en/interfaces/formats#csv

Reply:
need to make sure a general CSV reader can not rely on knowledge like` the second field is a JSON`  to parse a CSV.
without this knowledge,   a CSV reader at least knows the ` the second field is a string` and can parse it accordingly, and leave to upper level to decode the string further ( e.g. decode to nested structs).

Reply:
I understand what you mean, reading as a `CSV` string format works for me but I'm not sure the `,` is enclosed with a quote in the user's `CSV` file, we need some sample test files to check it.

Reply:
if `,` is neither enclosed or escaped, it`s invalid, we don`t need to consider it .

on the other side, normally the string must be quoted, and the the `"` is escaped as `""`(or "\"" for some formatter) 
and will leave to error whey you try to decode it as a normal JSON.

the CSV  format has to follow some rules after all.